### ISSUE TITLE
Deep Image Prior

### ISSUE THREAD

**Issue by reyllama** [reyllama/paper-reviews#81]
*CVPR 2018*

## Motivation

- Does Neural Network **Architecture** impose strong prior?
- Is the output Image space of a parameterized function (NN) significantly constrained?

## Key Idea

- Deep neural networks are expressive enough to *overfit* any signal.
- Network structure constrains training trajectory.
- By early stopping, we can reach unintended *denoising*.

## Method

![image](https://user-images.githubusercontent.com/46308407/127643816-249e0acc-2c94-450b-b1e9-aff7728d7be6.png)

- Fix a random noise (e.g. U[-1,1]) and gradient update network parameters under pixel-wise L2 distance.
- Note that in the end, the network restores *noisy* image quite well. That is because big enough NN are universal function approximator.
- However, in the trajectory, NN structure favors *natural, structured* image.

![image](https://user-images.githubusercontent.com/46308407/127643991-0377114d-e61c-449c-879f-bc38592e7c02.png)
- NN output image first approaches **clean, structured, natural** image and then learns to fit noise afterwards.
- This means Convolutional NNs favor structured, natural images over noise. (=prior)

![image](https://user-images.githubusercontent.com/46308407/127644118-c064c3f3-ea9a-4f8b-92f1-348eb26a95c3.png)
- Above graph shows that it takes much longer to fit random noise, compared to natural image.
- Although NNs can fit any noise in the end, in early training stages it reaches clean images.

![image](https://user-images.githubusercontent.com/46308407/127644210-70898d13-a947-4767-8fe3-a5f98704a115.png)
- This deep image prior can be employed in denoising, super-resolution and inpainting.
- By early stopping, we can obtain rather clean, natural image from noisy imput.

### Denoising
![image](https://user-images.githubusercontent.com/46308407/127644478-4db6f4cc-f426-4228-b203-84baae7bd8d7.png)
![image](https://user-images.githubusercontent.com/46308407/127644318-07609a95-4bff-49e1-85eb-d8ea790338f6.png)

### Super-resolution
![image](https://user-images.githubusercontent.com/46308407/127644340-3a24c139-5ae4-4675-8c99-ea5b1f6f9c0c.png)

### Inpainting
![image](https://user-images.githubusercontent.com/46308407/127644381-339794c4-bbaf-49e3-a6d3-e7cdd8cef4b9.png)

## Comments
- Very interesting idea to study the *prior* of convolutional neural networks.
- The observation that deep neural networks tend to favor natural images look magical, but in fact is related to **intrinsic dimension** of image manifold.
- Natural images reside in lower dimensional manifold, as there are many *constraints* for an image to be *natural*.
- Humans, for example, has to have certain range of skin tone, mostly two eyes, a nose, mouth, and fluffy object on top (hair).
- So, I think it's not that deep neural networks favor *natural* images, but they learn images of *low intrinsic dimension* first and then develop more complex representations.
- We know that pure noise has the highest entropy, thus highest intrinsic dimension. This is of course most difficult to learn/fit.
- However, how the authors used deep image priors to multiple **image restoration tasks** is interesting. Denoising is expected, but inpainting startles me, honestly speaking.
